172 implement static values for epoll

gwolf-011235 opened the pull request:
- Implemented s_maxEvents and s_epollTimeout and use them in main
- Also moved init of logger up to be able to log configfile path.
- Little bugfix: while checking for error in checkForTimeout (missing brackets) I moved setting of status Timeout too high. We never entered the CGI cases --> fixed
- add new integration test_timeout.py. Tries to simulate timeout. Since we don't have way to dynamically change timeout value these take very long (60 sec a pop). Added marker and deactivated timeout tests for "normal"  test run --> -m "not timeout" in pytest.ini. If removed will also run timeout tests
- added script infinite.sh, which just has infinite loop --> used for testing

u413-284-si reviewed (changes requested):
The infinite loop timeout test fails with a bad fd :(

gwolf-011235 commented:
I think this was just flaky test: we have timeout set at 60 seconds and I had test timeout at 61 seconds - so not much room for error. I set timeout in test to 65 (which means test take longer but shouldn't fail)

I'm pretty sure bad fd came from epoll trying to remove CGI file descriptor, which is still registered. Which didn't make test fail, but was just the last thing server reported.
I changed webutils::closeFd to removeCGIFiledescriptor() --> I think we can remove them at timeout stage (if they exist)
I also adapted epoll remove event to only send WARN message, so it doesn't look so scary.
And I init time string buffer to zero - not sure if it makes a difference but shouldn't hurt either.

u413-284-si commented:
Looks good now. Just there should be a better way to activate the timeout tests other than deleting the marker from pytest.ini, no?

gwolf-011235 commented:
- removed addopts from pytest.ini
- added automatic marker generation based on test name to conftest.py
- added markers to pytest.ini
- added variable PYTEST to test2 which contains addopts with default value `-v -m "not timeout"`

We can change on cli: `make test2 PYTEST=`  to run all or `make test2 PYTEST="-m cgi"` to run all cgi tests (n.b.: needs quoting)
